compat/win32/pthread.c

```diff
@@ -57,6 +57,8 @@ pthread_t pthread_self(void)
 	return t;
 }
 
+#ifdef GIT_WIN_XP_SUPPORT
```

Comment:
@dscho suggested this solution in this [google groups post](https://groups.google.com/d/msg/git-for-windows/EC5Cu-OdUCA/sDEyrCSDAgAJ) (please also read the full thread).

I don't have a Windowx XP machine to test, nor do I care to support it. All I can do is to make sure git compiles with and without the define (at least for this PR).